examples/cmake-scenarios/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+message(STATUS "Building using CMake version: ${CMAKE_VERSION}")
+cmake_minimum_required(VERSION 3.12)
+
+project(NanoArrowExampleCMakeMinimal LANGUAGES C CXX)
+
```

Comment:
`std::optional` requires C++17, so that's probably what you're hitting. I don't recall using any other C++17 features offhand. I could check and see, but in any case we're stripping down the example so not a big deal.

I assumed C++17 was OK because [that's what arrow requires](https://arrow.apache.org/docs/developers/cpp/building.html#system-setup), but it's valuable for nanoarrow to support older systems.